internal/schemautil/userconfig/apiconvert/toapi.go

```diff
@@ -270,6 +270,57 @@ func itemToAPI(
 	return res, o, nil
 }
 
+// processManyToOneKeys is a function that processes many to one keys.
+func processManyToOneKeys(res map[string]interface{}, depthIdx int) {
+	// mto is a map of many to one keys that exist in the provided properties.
+	mto := map[string][]string{}
+
+	// TODO: Remove all ip_filter and namespaces special cases when these fields are removed.
+	for k, v := range res {
+		if va, ok := v.(map[string]interface{}); ok {
+			processManyToOneKeys(va, depthIdx+1)
+		}
+
+		// We ignore untyped keys, because they cannot be many to one.
+		if !userconfig.IsKeyTyped(k) && k != "ip_filter" && k != "namespaces" {
+			continue
+		}
+
+		// rk is the real key, i.e. the key without the suffix.
+		var rk string
+
+		if k != "ip_filter" && k != "namespaces" {
+			rk = k[:strings.LastIndexByte(k, '_')]
+		} else {
+			rk = k
+		}
+
+		if _, ok := mto[rk]; !ok {
+			mto[rk] = []string{}
+		}
+
+		mto[rk] = append(mto[rk], k)
+	}
+
+	for k, v := range mto {
```

Comment:
It seems to me that we can simplify this function by removing this additional for and handling new values in the array from above. WDYT?

Reply:
I checked further, looks like this can't be simplified any further, but thanks for the heads up!